File: 0575-distribute-candies/0575-distribute-candies.java
class Solution {
    public int distributeCandies(int[] candies) {
	int total = candies.length;
	Set<Integer> kinds = new HashSet<Integer>();
	for (int i = 0; i < total; i++) {
		kinds.add(candies[i]);
	}
	if (kinds.size() > total / 2) {
		return total / 2;
	} else {
		return kinds.size();
	}
}
}

File: 0575-distribute-candies/0575-distribute-candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candyType) {
        sort(candyType.begin(),candyType.end());
        int count = 1;
        for(int i = 0;i<candyType.size();i++){
            for(int j = i+1;j<candyType.size();j++){
                if(candyType[i]==candyType[j]){
                    continue;
                }
                else if(candyType[i]!=candyType[j]){
                    i=j;
                    i--;
                    count++;
                    break;
                }
            }
        }
        int n = candyType.size()/2;
        if(count == n){
            return n;
        }
        else{
            return min(count,n);
        }
    }
};